kgforge/core/commons/sparql_query_builder.py

```diff
@@ -169,20 +169,15 @@ def build(
                     value = format_type[value_type](
                         parsed_value if parsed_value else f.value
                     )
-                    if value_type is CategoryDataType.LITERAL:
-                        if f.operator not in ["__eq__", "__ne__"]:
-                            raise NotImplementedError(
-                                "supported operators are '==' and '!=' when filtering with a str."
-                            )
-                        statements.append(f"{property_path} ?v{index}")
-                        sparql_filters.append(
-                            f"FILTER(?v{index} = {_box_value_as_full_iri(value)})"
-                        )
-                    else:
-                        statements.append(f"{property_path} ?v{index}")
-                        sparql_filters.append(
-                            f"FILTER(?v{index} {sparql_operator_map[f.operator]} {_box_value_as_full_iri(value)})"
+                    if value_type is CategoryDataType.LITERAL and f.operator not in ["__eq__", "__ne__"]:
```

Comment:
do we have a test for this? checking if it is a string, it should only accept == or !=

Reply:
There is a test in https://github.com/BlueBrain/nexus-forge/blob/master/tests/specializations/stores/test_bluebrain_nexus.py#L560

Reply:
I am wandering though why this test were passing. Looking at the change it might mean we did not have a Literal value type in the test. @ssssarah can add that case in the test ?